contributing-docs/07_local_virtualenv.rst

```diff
@@ -61,26 +62,114 @@ of required packages.
    released wheel packages.
 
 
-Installing Airflow
-..................
+Creating and maintaining local virtualenv with uv
+-------------------------------------------------
+
+As of November 2024 we are recommending to use ``uv`` for local virtualenv management for Airflow development.
+Th e``uv`` is a build frontend tool that is designed to manage python, virtualenvs and workspaces for development
+and testing of Python projects. It is a modern tool that is designed to work with PEP 517/518 compliant projects
+and it is much faster than "reference" ``pip`` tool. It has extensive support to not only create development
+environment but also to manage python versions, development environments, workspaces and Python tools used
+to develop Airflow (via ``uv tool`` command - such as ``pre-commit`` and others, you can also use ``uv tool``
+to install ``breeze`` - containerized development environment that we use to be able to reproduce the
+CI environment locally and to run release-management and certain development tasks.
+
+You can read more about ``uv`` in `UV Getting started <https://docs.astral.sh/uv/getting-started/>`_ but
+below you will find a few typical steps to get you started with ``uv``.
+
+Installing uv
+.............
+
+You can follow the `installation instructions <https://docs.astral.sh/uv/getting-started/installation/>`_ to install
+``uv`` on your system. Once you have ``uv`` installed, you can do all the environment preparation tasks using
+``uv`` commands.
+
+Installing Python versions
+..........................
+
+You can install Python versions using ``uv python install`` command. For example, to install Python 3.9.7, you can run:
+
+.. code:: bash
+
+    uv python install 3.9.7
+
+This is optional step - UV will automatically install the Python version you need when you create a virtualenv.
+
+Creating virtualenvs with uv
+............................
+
+.. code:: bash
+
+    uv venv
```

Comment:
How does it otherwise detect the version?

Reply:
It uses default version that the project support (from pyproject.toml) - which is Python 3.12. That's the default behaviour of `uv`. If you think it's worth adding it, feel free :).

Reply:
Not needed, sounds good like this

Reply:
It was my curiosity that asked this :)